art/attacks/inference/membership_inference/label_only_boundary_distance.py

```diff
@@ -70,6 +71,18 @@ def infer(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.n
         """
         from art.attacks.evasion.hop_skip_jump import HopSkipJump
 
+        if y is None:
+            raise ValueError("Argument `y` is None, but this attack requires true labels `y` to be provided.")
+
+        if self.distance_threshold_tau is None:
+            raise ValueError(
+                "No value for distance threshold `distance_threshold_tau` provided. Please set"
+                "`distance_threshold_tau` or run method `calibrate_distance_threshold` on known training and test"
+                "dataset."
+            )
+
+        y = check_and_transform_label_format(y, self.estimator.nb_classes)
```

Comment:
Call to check_and_transform_label_format should also be added to calibrate_distance_threshold method (both on y_train and y_test)

Reply:
I agree and have updated the code.

Reply:
Ok. Looks good.